src/scripting_api/app.js

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EventDispatcher } from "./event.js";
+import { NotSupportedError } from "./error.js";
+import { PDFObject } from "./pdf_object.js";
+
+class App extends PDFObject {
+  constructor(data) {
+    super(data);
+    this._specialId = true;
```

Comment:
It's not really clear what this does, and it's always `true`.

Reply:
It's clearly not meaningful. So I changed it to `_isApp`, this field is used in `proxy.js` when checking for the backdoor.
This backdoor is required (I didn't find any other way to do) to call a function (`App::_dispatchEvent`) which is not in the JS api (see PdfSandbox.jsm at line 72 in m-c patch).